cardano-db-sync/src/Cardano/DbSync/AppT.hs

```diff
@@ -0,0 +1,294 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Cardano.DbSync.AppT (
+  AppT (..),
+  App,
+  MonadAppDB (..),
+  runApp,
+  runAppInIO,
+  convertAppToReaderT,
+  askTrace,
+  getTraceFromSyncEnv,
+  askNetwork,
+  runAppWithNoLogging,
+  runAppWithLogging,
+  askInsertOptions,
+  liftAtomicallyT,
+  NoLedgerEnv (..),
+  SyncEnv (..),
+  SyncOptions (..),
+  InsertOptions (..),
+  LedgerEnv (..),
+  HasLedgerEnv (..),
+  StateQueryTMVar (..),
+  FixesRan (..),
+  ConsistentLevel (..),
+  EpochState (..),
+  RunMigration,
+)
+where
+
+import Cardano.BM.Trace (Trace, logError)
+import qualified Cardano.Db as DB
+import Cardano.DbSync.Cache.Types (CacheState)
+import Cardano.DbSync.Config.Types (LedgerStateDir, SyncNodeConfig)
+import Cardano.DbSync.Error.Types (SyncNodeError)
+import Cardano.DbSync.Ledger.Types (CardanoLedgerState, LedgerDB)
+import Cardano.DbSync.Types (CardanoBlock, CardanoInterpreter, OffChainPoolResult, OffChainPoolWorkQueue, OffChainVoteResult, OffChainVoteWorkQueue)
+import qualified Cardano.Ledger.Shelley.API as Ledger
+import Cardano.Prelude (
```

Comment:
If we're using `NoImplicitPrelude`, we might as well just import all of `Cardano.Prelude`. Better yet, we might consider adding it our custom prelude implicitly with [mixins](https://kowainik.github.io/projects/relude#mixins)

Reply:
yeah I agree, there are lots of places where it uses prelude and others `Cardano.Prelude` I'm happy with the later and having `NoImplicitPrelude` globally.